### ISSUE TITLE
Investigate expired sessions can still register

### ISSUE THREAD

**Issue by mnelson4** [eventespresso/event-espresso-core#2281]
## Issue Overview

While testing https://github.com/eventespresso/event-espresso-core/pull/2060, Josh noted:
> I also tested what would happen after one hour when the transaction's AHEE__EE_Cron_Tasks__expired_transaction_check was fired and found that the reserved counts were cleared, but that I could still complete the transaction as it seemed to still have an open session. Which could lead to overselling if someone else happened to register in between the time the AHEE__EE_Cron_Tasks__expired_transaction_check was run and the other transaction was completed. This is also an issue on master

## Bug report or feature request?

* [x] Bug
* [ ] Feature
* [ ] Neither

## Environment Data:

Version of EE: 
4.10.3
Version of WordPress:-

PHP Version:  -

Browser used:  -

## Steps to Reproduce (for bugs)

* Begin a registration
* Wait over an hour
* try to continue registering

## Expected Behaviour

You shouldn't be able to complete the registration.

## Current Behaviour

You can complete the registration, which could lead to overselling.

**Comment by mnelson4** [comment]
I've tentatively assigned myself to investigate, but it's a-ok to pass on to someone else!

**Comment by mnelson4** [comment]
I've tried reproducing this but couldn't. 
I created a free event and a paid one. I began a registration for the free one in Firefox (just submitted the ticket quantity, but didn't enter any attendee details), the paid one in Chrome (entered attendee details but didn't pay). 
Waited over an hour until the action fired by itself. 
The reservations were released ok on the ticket, and I couldn't complete either transaction (each gave a different error). So I'm not sure if I'm doing something differently or not.
